app/models/katello/repository.rb

```diff
@@ -680,6 +680,49 @@ def docker_meta_tag_count
       DockerMetaTag.in_repositories(self.id).count
     end
 
+    # a master repository actually has content (rpms, errata, etc) in the pulp repository.  For these repositories, we can use the YumDistributor
+    # to generate metadata and can index the content from pulp, or for the case of content view archives without filters, can also use the YumCloneDistributor
+    #
+    def master?
+      !self.yum? || # non-yum repos
+          self.in_default_view? || # default content view repos
+          (!self.in_default_view? && self.archive? && !self.content_view.composite) || # non-composite content view archive repos
+          (self.content_view.composite? && self.component_source_repositories.count > 1) # composite archive repo with more than 1 source repository
+    end
+
+    # a link repository has no content in the pulp repository and serves as a shell.  It will always be empty.  Only the YumCloneDistributor can be used
+    # to publish yum metadata, and it cannot be indexed from pulp, but must have its indexed associations copied from another repository.
+    def link?
+      !master?
+    end
+
+    def target_repository
```

Comment:
Add a comment along the lines of  "a link points to a target". While it sounds obvious from name I think a line here will also reinforce that idea

Reply:
May be also add comments here to say things like ```Archived versions in a non-composite content view will always be master. However in the case of a composite content view they'd be links```